src/os/bluestore/BlueStore.cc

```diff
@@ -14516,12 +14560,14 @@ void BlueStore::_do_truncate(
   if (offset == o->onode.size)
     return;
 
+  int64_t ondisk_offset = 0;
   if (offset < o->onode.size) {
     WriteContext wctx;
     uint64_t length = o->onode.size - offset;
     o->extent_map.fault_range(db, offset, length);
     o->extent_map.punch_hole(c, offset, length, &wctx.old_extents);
     o->extent_map.dirty_range(offset, length);
+    ondisk_offset = wctx.old_extents.begin()->r.begin()->offset;
```

Comment:
IMO this might be wrong when old_extents is emplty, e.g. when the trimmed tail is shorted than last extent. Highly likely @tchaikov's QA run failure in https://github.com/ceph/ceph/pull/36230 is caused by the case

Reply:
@ifed01 thanks for catching this!  Can you please take a look at the fix?  Thanks!

Reply:
LGTM